xla/service/cpu/onednn_config.proto

```diff
@@ -0,0 +1,111 @@
+syntax = "proto3";
+
+package xla.cpu;
+
+message OneDnnDataLayoutProto {
+  // The batch dimension of the tensor
+  uint64 batch_dim = 1;
+  // The feature dimension of the tensor
+  uint64 feature_dim = 2;
+  // The spatial dimensions of the tensor
+  repeated uint64 spatial_dims = 3; 
+}
+
+message OneDnnFilterLayoutProto {
+  // The input feature dimension of the tensor
+  uint64 input_feature_dim = 1;
+  // The output feature dimension of the tensor
+  uint64 output_feature_dim = 2;
+  // The spatial dimensions of the tensor
+  repeated uint64 spatial_dims = 3;
+  // Shape of the tensor
+  repeated uint64 shape = 4; 
+}
+
+message OneDnnFactorLayoutProto {
+  // The dimensions of the tensor
+  repeated uint64 dimensions = 1;
+  // Shape of the tensor
+  repeated uint64 shape = 2;  
+}
+
+message OneDnnOptimizationConfig {
+  bool weights_prepacked = 1;
+  bool user_scratchpad = 2;
+  bool bias_broadcast = 3;
+}
+
+message OneDnnFusionConfig {
+  // These enum needs to be mapped to oneDNN enum for post_op algorithm.
+  // TODO(intel-tf): Add kinds supported by oneDNN.
+  enum FusionKind {
+    UNDEFINED = 0;
+    BIAS = 1;
+    RELU = 2;
+    TANH = 3;
+    GELU_ERF = 4;
+    GELU_TANH = 5;
+    BINARY_ADD = 6;
+    LINEAR = 7;
+    ELU = 8;
+    RELU6 = 9;
+    SIGMOID = 10;
+  }
+  repeated FusionKind ops = 1;
+  // To avoid protobuf failures for specific decimal values,
+  // the original float value alpha is type-casted to int32.
+  int32 alpha_typecast = 2;
+}
+
+message OneDnnTensorLayoutProto {
+  uint64 dims = 1;
+  oneof layout {
+    OneDnnDataLayoutProto data = 3;
+    OneDnnFilterLayoutProto filter = 4;
+    OneDnnFactorLayoutProto tensor = 5;
+  }
+}
+
+message OneDnnSoftmaxConfig {
+  int32 softmax_axis = 1;
+}
+
+message OneDnnMatMulConfig {
+  bool transpose_a = 1;
+  bool transpose_b = 2;
+
+  OneDnnFusionConfig fusions = 3;
+  OneDnnOptimizationConfig optimization_config = 8;
```

Comment:
4 instead of 8?

Reply:
Currently, in the main branch OneDnnMatmulConfig already has ids from 1 through 7 in use, and in the commits to proto files in the repo I see authors don't re-use a previously used id so I thought to give new items new and unique ids.

Reply:
This is a new message, so new ordering can be used. Only within one message, different ids need to be used. I am not aware of people actively avoiding to reuse ids from different messages.

Reply:
Here, we refactor a lot of stuff out from OneDnnMatmulConfig. I think because in this PR we moved it to a new file it shows up as a new message, but we do have this message in the main branch. This is what it looks like in main right now:

```
message OneDnnMatMulConfig {
  bool transpose_a = 1;
  bool transpose_b = 2;
  // These enum needs to be mapped to oneDNN enum for post_op algorithm.
  // TODO(intel-tf): Add kinds supported by oneDNN.
  enum FusionKind {
    UNDEFINED = 0;
    BIAS = 1;
    RELU = 2;
    TANH = 3;
    GELU_ERF = 4;
    GELU_TANH = 5;
    BINARY_ADD = 6;
    LINEAR = 7;
    ELU = 8;
    RELU6 = 9;
    SIGMOID = 10;
  }
  repeated FusionKind fused_ops = 3;
  bool bias_broadcast = 4;
  // To avoid protobuf failures for specific decimal values,
  // the original float value alpha is type-casted to int32.
  int32 alpha_typecast = 5;
  bool weights_prepacked = 6;
  bool user_scratchpad = 7;
}
```

Reply:
Ok, I understand now that this is just a proto message that got moved from backend_config.proto to onednn_config.proto. In that case, I would suggest adding "reserved" tags for those fields which will not exist anymore after the move.
e.g.
```
reserved 4;  // Was bias_broadcast
```